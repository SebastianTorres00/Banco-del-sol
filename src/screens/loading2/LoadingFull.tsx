import React, { useEffect, useState } from 'react';
import { View, Animated, Easing } from 'react-native';
import { useSelector } from 'react-redux';
import { Props, resFromBack } from '../../types/Types';
import { styles } from './LoadingFullStyles';

export function LoadingFull({ navigation }: Props) {
  const userStore = useSelector((state: resFromBack) => state.user);
  const [spinAnim, setSpinAnim] = useState(new Animated.Value(0));
  const spin = spinAnim.interpolate({
    inputRange: [0, 1],
    outputRange: ['0deg', '3600deg'],
  });

  useEffect(() => {
    Animated.loop(
      Animated.timing(spinAnim, {
        toValue: -1,
        duration: 60000,
        easing: Easing.linear,
        useNativeDriver: true,
      }),
    ).start();
  });

  return (
    <View style={styles.container}>
      <Animated.Image
        style={{ height: 150, width: 150, transform: [{ rotate: spin }] }}
        source={require('../../../assets/Banco-del-Sol-Logo sol.png')}
      />
      {userStore.email && userStore.email.length
        ? navigation.push('Home')
        : null}
    </View>
  );
}
