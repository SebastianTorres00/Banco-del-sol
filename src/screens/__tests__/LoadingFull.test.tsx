import React from 'react';
import { shallow } from 'enzyme';
import { LoadingFull } from '../loading2/LoadingFull';

describe('Loading', () => {
  it('renders the loading', () => {
    const wrapper = shallow(<LoadingFull />);
    expect(wrapper).toMatchSnapshot();
  });
});
