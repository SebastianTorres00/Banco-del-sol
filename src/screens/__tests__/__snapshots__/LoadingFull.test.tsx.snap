// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders the text 1`] = `ShallowWrapper {}`;
